python_modules/dagster/dagster/_core/definitions/asset_graph.py

```diff
@@ -255,10 +250,9 @@ def get_execution_set_asset_and_check_keys(
         if isinstance(asset_or_check_key, AssetKey):
             return self.get(asset_or_check_key).execution_set_asset_and_check_keys
         else:  # AssetCheckKey
-            asset_node = self._asset_nodes_by_check_key[asset_or_check_key]
```

Comment:
fetching the asset node doesn't make much sense here- the check might be in a different op